rotors_gazebo_plugins/src/gazebo_odometry_plugin.cpp

```diff
@@ -236,47 +257,69 @@ void GazeboOdometryPlugin::OnUpdate(const common::UpdateInfo& _info) {
   }
 
   if (gazebo_sequence_ % measurement_divisor_ == 0) {
-    nav_msgs::Odometry odometry;
-    odometry.header.frame_id = parent_frame_id_;
-    odometry.header.seq = odometry_sequence_++;
-    odometry.header.stamp.sec = (world_->GetSimTime()).sec + ros::Duration(unknown_delay_).sec;
-    odometry.header.stamp.nsec = (world_->GetSimTime()).nsec + ros::Duration(unknown_delay_).nsec;
-    odometry.child_frame_id = child_frame_id_;
-    copyPosition(gazebo_pose.pos, &odometry.pose.pose.position);
-    odometry.pose.pose.orientation.w = gazebo_pose.rot.w;
-    odometry.pose.pose.orientation.x = gazebo_pose.rot.x;
-    odometry.pose.pose.orientation.y = gazebo_pose.rot.y;
-    odometry.pose.pose.orientation.z = gazebo_pose.rot.z;
-    odometry.twist.twist.linear.x = gazebo_linear_velocity.x;
-    odometry.twist.twist.linear.y = gazebo_linear_velocity.y;
-    odometry.twist.twist.linear.z = gazebo_linear_velocity.z;
-    odometry.twist.twist.angular.x = gazebo_angular_velocity.x;
-    odometry.twist.twist.angular.y = gazebo_angular_velocity.y;
-    odometry.twist.twist.angular.z = gazebo_angular_velocity.z;
+
+    gz_geometry_msgs::Odometry odometry;
+    odometry.mutable_header()->set_frame_id(parent_frame_id_);
```

Comment:
sorry was a bit unclear. I thought of a function that converts a gazebo header to a ros header since all headers have the same structure and you could safe a lot of code duplication